include/caf/net_state.rst

```diff
@@ -0,0 +1,38 @@
+.. _caf_net_state:
+
+CAF: Network state
+##################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |net_state| of the :ref:`lib_caf` (CAF) is responsible for broadcasting the events about current network state.
+The module provides different backends for available networks (like LTE or OpenThread).
+
+Configuration
+*************
+
+You can enable |net_state| by selecting the :option:`CONFIG_CAF_NET_STATE` option in the configuration.
+The module selects the backend based on the link layer enabled.
+
+
+Log information about connection waiting
+========================================
+
+The option :option:`CONFIG_CAF_NET_STATE_WAITING` enables periodic message logged while the module is waiting for network connection.
+The period between logs may be configured by :option:`CONFIG_CAF_LOG_NET_STATE_WAITING_PERIOD`.
+
+Implementation details
+**********************
+
+The module generates the ``net_state_event``.
+The event reports current network state using :c:enum:`net_state`.
+Following network states are available:
+
+* NET_STATE_DISABLED - Network interface is disabled
+* NET_STATE_DISCONNECTED - Network is ready to connect but waiting for connection
```

Comment:
```suggestion
* :c:enum:`NET_STATE_DISCONNECTED` - Network interface is ready to connect but waiting for connection.
```
Are there any conditions to be met for connection to take place (wait to end)? Any event that needs to be received? Would it make sense to mention them here?

Reply:
It would not make any sense here. It is informing module. Not a module with takes care about it how to initialize network to be able to connect. It may vary depending on the backend.